color/regex.c

```diff
@@ -231,11 +231,14 @@ static enum CommandResult add_pattern(struct RegexColorList *rcl, const char *s,
     // different colours
     if (cc && ((cc->fg != fg) || (cc->bg != bg)))
     {
-      attr_color_clear(&rcol->attr_color);
       cc = curses_color_new(fg, bg);
```

Comment:
Would it be unreasonable to handle unset/default too, so we're sure that this function always return a valid pointer?

Reply:
The `CursesColor` maps to an ncurses colour in the terminal.
We're only allowed `COLORS` (256) of those.
`cc->index` tells us which we've used.

If we allocated empty `CursesColor`s, we'd just make allocating them more complicated.

(I'll add colours to the glossary :-)

Reply:
Can't we have a default one we use as a fallback? Or you'll have to check every call.. e.g., what happens at line 284?

Reply:
At L284, we don't care whether there's a colour, or not.

If `ac->curses_color` is non-`NULL`, then we have a colour.
This is tested in quite a few place, notably when we want to merge colours.

Having an empty `CursesColor` would mean something like:
```diff
-  if (ac->curses_color)
+  if (ac->curses_color != CurseColorEmpty)
```